### PR TITLE
Native testing

### PR DESCRIPTION
**Pull request by kadaan**

### CODE REVIEW

**Comment by kadaan** [comment]
@bau-sec Automated testing!  My feeling is we should get this in and then put the CPS tests in as native tests.

**Comment by bau-sec** [comment]
> 
> 
> @bau-sec Automated testing! My feeling is we should get this in and then put the CPS tests in as native tests.

That's fine with me, no rush on the CPS PR. The CPS unit tests fixed some things, but they were all minor and/or edge cases, nothing that was affecting my actual rides.

File: lib/SS2K/src/sensors/FlywheelData.cpp [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier: GPL-2.0-only
  */
 
-#include "BLE_Common.h"
+#include "os/endian.h"
```
Comment (doudar) [inline lib/SS2K/src/sensors/FlywheelData.cpp:8]: Endian lib is pretty nice - and small. Good find.

File: platformio.ini [hunk 2]
```diff
@@ -7,16 +7,20 @@
 ;
 ; Please visit documentation for the other options and examples
 ; https://docs.platformio.org/page/projectconf.html
+[platformio]
+default_envs = esp32doit
+
 [common_env_data]
 lib_deps = 
-    ;https://github.com/h2zero/NimBLE-Arduino.git#master //Nimble is tracked and including this will probably cause issues.
```
Comment (doudar) [inline platformio.ini:12]: Should probably keep this here? I'm guessing we will eventually add it back in?

Comment (doudar) [reply platformio.ini:12]: Or is your plan to go completely cross platform dependent?

File: lib/SS2K/include/Constants.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * All rights reserved
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#pragma once
+
+#include <NimBLEUUID.h>
+
+#define CYCLINGPOWERMEASUREMENT_UUID NimBLEUUID((uint16_t)0x2A63)
```
Comment (doudar) [inline lib/SS2K/include/Constants.h:12]: I'm perfectly okay with moving the UUID's out of BLE common.h but not okay with splitting them up. They all need to be moved here or left as a group over there.

File: platformio.ini (hunk 2 above)
Comment (kadaan) [reply platformio.ini:12]: Nope

File: lib/SS2K/include/Constants.h (hunk 3 above)
Comment (kadaan) [reply lib/SS2K/include/Constants.h:12]: Okay. I was looking at this as an interim step. I’m fine moving them all here now. I think much of the advertising code moves to the lib, so it can be easily tested.

**Comment by kadaan** [comment]
Still need to ride on SS2K with this change.

**Review by doudar — APPROVED** [review]

**Review by doudar — APPROVED** [review]
